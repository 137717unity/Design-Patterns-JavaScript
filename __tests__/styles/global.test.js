import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { themeLight } from '../../src/styles/themes/theme.light';
import { themeDark } from '../../src/styles/themes/theme.dark';
import GlobalStyle from '../../src/styles/global';

describe('Global style', () => {
  it('renders with a LIGHT theme', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={themeLight}>
          <body>
            <GlobalStyle />
            <div id="root">TEST</div>
          </body>
        </ThemeProvider>
      )
      .toTree();

    expect(
      tree.instance._reactInternalFiber.child.child.child.child.stateNode.styleSheet.target
    ).toMatchSnapshot();
  });

  it('renders with a DARK theme', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={themeDark}>
          <body>
            <GlobalStyle />
            <div id="root">TEST</div>
          </body>
        </ThemeProvider>
      )
      .toTree();

    expect(
      tree.instance._reactInternalFiber.child.child.child.child.stateNode.styleSheet.target
    ).toMatchSnapshot();
  });
});
