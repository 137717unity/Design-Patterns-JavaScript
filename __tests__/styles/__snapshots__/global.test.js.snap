// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global style renders with a DARK theme 1`] = `
<head>
  <style
    data-styled=""
    data-styled-version="4.1.3"
  >
    
    
/* sc-component-id: sc-global-2479929527 */
body{background:#222222;font-family:'Karla',sans-serif;margin:0;padding:0;} #root{max-width:640px;margin:auto;}
  </style>
</head>
`;

exports[`Global style renders with a LIGHT theme 1`] = `
<head>
  <style
    data-styled=""
    data-styled-version="4.1.3"
  >
    
    
/* sc-component-id: sc-global-2479929527 */
body{background:#f4f4f4;font-family:'Karla',sans-serif;margin:0;padding:0;} #root{max-width:640px;margin:auto;}
  </style>
</head>
`;
